### DISCUSSION TITLE
[Navigation] - Navigate Across All The Dashboards Directly From Any Where In The App

### DISCUSSION THREAD

**Discussion by pankajsoni22** [apache/superset#20897] [Ideas]
# Purpose of discussion:
This discussion has a purpose of improving user experience in terms of switching between different dashboard or want to land on a particular dashboard from anywhere in the app.

# Issue with current design:
Current design does not provide one click redirection to any dashboard if user is not on `/dashboard/list` page. User has to go to '/dashboards/list' page first and then select from there, as you can see in the following diagram current design is
![superset-nav-old](https://user-images.githubusercontent.com/39917736/181445578-3cd25ecb-b319-40b3-8eb6-8a856ec91ad4.png)

Cons:
1. It takes more time for the user to land on dashboard if he/she is working some where in explore/sql editor/saved queries.
2. It waste resources on client side (like bandwidth), page rendering.
3. Every time going through dashboard list makes unnecessary calls to backend.

# Proposed solution:
The solution gives an idea towards a new item ingestion in menu i.e. Navigate which holds list of all current dashboards allowed to particular user, these all items are links to dashboards and with one click user will be directly landing on corresponding dashboard he/she selected. Proposed solution has the following flow and design changes
![superset-nav-new](https://user-images.githubusercontent.com/39917736/181448477-4b724fdc-6d86-4f21-98f8-54fdd54a4cc0.png)

Pros:
1. Bypass dashboard list page as no more required to get list of dashboards - this saves time of user.
2. This saves client side resources as this is a quick landing without useless rendering of dashboard list page.
3. Bypassing dashboard list page will naturally reduce calls to backend.

Above solution on typical superset dashboard may look like following screenshots:
1. Introduction of one more item in top-menu: 
![super-nav-dropdown-highlight](https://user-images.githubusercontent.com/39917736/181449003-a3502e92-1adf-4f21-b3ed-90a9491d37d9.png)

2. List of dashboard in top-menu from `dashboard/list` page:
![super-nav-from-dashboard-list](https://user-images.githubusercontent.com/39917736/181449453-c84052bf-7e5a-4bb6-a293-3b065e9a73f1.png)

3. List of dashboard in top-menu from `superset/lsqllab` page:
![super-nav-from-sql-editor](https://user-images.githubusercontent.com/39917736/181449662-5e14ea3c-4a4c-4144-befe-135fc0d4a228.png)

4. List of dashboard in top-menu from `chart/list` page:
![super-nav-from-charts-list](https://user-images.githubusercontent.com/39917736/181449748-c9a44c44-8f9e-4e98-970d-d64fd1f994d7.png)

5. List of dashboard in top-menu from `databaseview/list` page:
![super-nav-from-databases](https://user-images.githubusercontent.com/39917736/181450083-8e280dc9-8f7d-420b-9f71-a7fcdbe1b7b0.png)

6. Switch among dashboards:
![super-nav-from-within-dashboard](https://user-images.githubusercontent.com/39917736/181450285-1ff3d753-1673-4e2a-9bed-b61110ce950a.png)

# Final Comments:
Superset community may allow or reject this design but the idea should be accepted, design may change as everytime going to dashboard list page is a hactic for user.

Please comment you thoughts and let me know if any design changes are required or this easy but impactable solution works.

**Comment by nytai** [comment]
hmmm, the same could be said about charts or saved queries, why limit to only dashboards? I think a search or jump to bar would be better, similar to slack's `cmd/ctrl + k` feature. It could then just list all the recently visited objects (like the home page currently does) when the search box is empty

**Reply by pankajsoni22** [reply]
Why to limit the feature to just recently used items ?
Users are more interested in dashboards as he/she wants to read collective information and switch among them, though I agree feature with saved queries but that can come in next iterations (keeping agile development in mind). On the other side we need to provide a simple UI and visible feature, which takes me to menu > dropdown(navigate).

**Reply by pankajsoni22** [reply]
And this can be a feature flag enabled feature.

**Reply by nytai** [reply]
Yup, I meant by default list the recently visited objects. Search would then be the way to find something more specific.

**Comment by kamalkeshavani-aiinside** [comment]
We had `OMNIBAR`, with similar but limited features than requested here. But it was removed in 2.0
https://github.com/apache/superset/pull/19241

**Comment by michael-s-molina** [comment]
Tagging @jess-dillard @kasiazjc for awareness.

**Comment by cccs-tom** [comment]
Is the main use case here to navigate to a related dashboard? Or something completely unrelated? If we are talking about related dashboards, I think that the ongoing work with drill actions should mostly cover this.

Having said that, if you're looking for a "Quick Jump" experience, I think there should be some sort of search involved. I'm thinking of something like the upper-left search bar in GitHub that searches all types of entities for the given search terms. Admittedly, that has a lot in common with the OMNIBAR. Or at least what I could glean of it from the link above - I've never used it myself...

**Reply by pankajsoni22** [reply]
This is about completely unrelated dashboards, the idea of left search bar is interesting.
_the ongoing work with drill actions should mostly cover this_  : which on going work you (@cccs-tom ) are talking about, can you please direct me to the same if any SIP/Discussion going on that ?

**Reply by cccs-tom** [reply]
Ok, so 'ongoing' might have been a bit of an over-statement here. The work on drill actions (and specifically "drill-across", which is what I was referring to earlier) is still in its infancy. For the most part, design discussions have started happening, but that's it. The SIP is still being drafted. That said, there is at least a [Slack channel](https://apache-superset.slack.com/archives/C03F4PPJB29) that you can join. It's not super active just yet, but hopefully that will change some time this Fall.

**Comment by hughhhh** [comment]
With this request I agree with @nytai that we should go with the cmd + k approach and essentially bring back the omnibar feature to search across all resources

**Reply by hughhhh** [reply]
Here's are all the omnibar related PRs:
https://github.com/pulls?q=is%3Apr+author%3Ahughhhh+archived%3Afalse+is%3Aclosed+omnibar

**Comment by mistercrunch** [comment]
+1 for a global search/omnibar approach. Also keep in mind that some Superset instances have thousands of dashboards!

**Reply by dpgaspar** [reply]
+1, we should evaluate using a search engine e.g.: elasticsearch
